restapi/src/main/java/io/stargate/web/docsapi/service/DocumentService.java

```diff
@@ -169,7 +169,17 @@ private boolean isEmptyArray(Object v) {
       boolean isJson) {
     String trimmed = payload.trim();
     if (isJson) {
-      return shredJson(surfer, db, path, key, trimmed, patching);
+      try {
+        return shredJson(surfer, db, path, key, trimmed, patching);
+      } catch (RuntimeException e) {
```

Comment:
Shouldn't we simply catch the concrete `RuntimeException` that you want to transform to the `ErrorCodeRuntimeException`? What s the exception thrown here?

Again I would catch a proper exception in `shredJson` based on the user input only (see other comment).

Reply:
So unfortunately, the exception thrown by `JsonSurfer` on malformed JSON is a plain RuntimeException, which is why I took this approach...since `shredJson` is really only managing user input, I chose to do the handling here...not ideal but there's not much for me to work with since the JsonSurfer API is throwing a plain RuntimeException, specificity is lost

Reply:
OK appraoch is fine then, as we can not really do anything about it.. Can we just move the whole to the method itself, then callers of the method don't have to care to repeat over and over the same procedure? Then I guess we can remove changes here and on the https://github.com/stargate/stargate/pull/1079/files#diff-caf69767609da0fee30d28b2b410ba5e400ed01e1f42af082f495ef82fec8416R494